contracts/andromeda-merkle-airdrop/src/contract.rs

```diff
@@ -0,0 +1,982 @@
+    STAGE_EXPIRATION.save(deps.storage, stage.into(), &exp)?;
+
+    // save total airdropped amount
+    let amount = total_amount.unwrap_or_else(Uint128::zero);
+    STAGE_AMOUNT.save(deps.storage, stage.into(), &amount)?;
+    STAGE_AMOUNT_CLAIMED.save(deps.storage, stage.into(), &Uint128::zero())?;
+
+    Ok(Response::new().add_attributes(vec![
+        attr("action", "register_merkle_root"),
+        attr("stage", stage.to_string()),
+        attr("merkle_root", merkle_root),
+        attr("total_amount", amount),
+    ]))
+}
+
+pub fn execute_claim(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    stage: u8,
+    amount: Uint128,
+    proof: Vec<String>,
+) -> Result<Response, ContractError> {
+    // not expired
+    let expiration = STAGE_EXPIRATION.load(deps.storage, stage.into())?;
+    require(
+        !expiration.is_expired(&env.block),
+        ContractError::StageExpired { stage, expiration },
+    )?;
+
+    // verify not claimed
+    require(
+        !CLAIM.has(deps.storage, (&info.sender, stage.into())),
+        ContractError::Claimed {},
+    )?;
+
+    let config = CONFIG.load(deps.storage)?;
+    let merkle_root = MERKLE_ROOT.load(deps.storage, stage.into())?;
+
+    let user_input = format!("{}{}", info.sender, amount);
+    let hash = sha2::Sha256::digest(user_input.as_bytes())
+        .as_slice()
+        .try_into()
+        .map_err(|_| ContractError::WrongLength {})?;
+
+    let hash = proof.into_iter().try_fold(hash, |hash, p| {
+        let mut proof_buf = [0; 32];
+        hex::decode_to_slice(p, &mut proof_buf)?;
+        let mut hashes = [hash, proof_buf];
+        hashes.sort_unstable();
+        sha2::Sha256::digest(&hashes.concat())
+            .as_slice()
+            .try_into()
+            .map_err(|_| ContractError::WrongLength {})
+    })?;
+
+    let mut root_buf: [u8; 32] = [0; 32];
+    hex::decode_to_slice(merkle_root, &mut root_buf)?;
+    require(root_buf == hash, ContractError::VerificationFailed {})?;
+
+    // Update claim index to the current stage
+    CLAIM.save(deps.storage, (&info.sender, stage.into()), &true)?;
+
+    // Update total claimed to reflect
+    let mut claimed_amount = STAGE_AMOUNT_CLAIMED.load(deps.storage, stage.into())?;
+    claimed_amount += amount;
+    STAGE_AMOUNT_CLAIMED.save(deps.storage, stage.into(), &claimed_amount)?;
+
+    let transfer_msg: CosmosMsg = match config.asset_info {
+        AssetInfo::Cw20(address) => CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: address.to_string(),
+            funds: vec![],
+            msg: to_binary(&Cw20ExecuteMsg::Transfer {
+                recipient: info.sender.to_string(),
+                amount,
+            })?,
+        }),
+        AssetInfo::Native(denom) => CosmosMsg::Bank(BankMsg::Send {
+            to_address: info.sender.to_string(),
+            amount: vec![Coin { amount, denom }],
+        }),
+    };
+
+    let res = Response::new()
```

Comment:
Not a change but rather just curiosity. Why do they instantiate this Response outside of the return?

Reply:
Just seems like personal preference, there aren't any performance differences or anything.